src/os/bluestore/BlueStore.cc

```diff
@@ -5397,6 +5397,17 @@ void *BlueStore::MempoolThread::entry()
     interval_stats_trim = false;
 
     store->refresh_perf_counters();
+    uint64_t period = store->cct->_conf.get_val<uint64_t>("bluestore_fragmentation_check_period");
+    if (period != 0) {
+      auto now = mono_clock::now();
+      timespan elapsed = now - last_fragmentation_check;
+      if (elapsed > make_timespan(period)) {
+        double score = store->alloc->get_fragmentation_score();
```

Comment:
Can we log at pretty low level (0?) before and after this call to be able to identify relevant stalls if any.